cds/modules/records/tasks.py

```diff
@@ -223,3 +228,108 @@ def file_integrity_report():
         recipients = [current_app.config['CDS_ADMIN_EMAIL']]
         msg = Message(subject, sender=sender, recipients=recipients, body=body)
         current_app.extensions['mail'].send(msg)
+
+
+def format_subformats_integrity_report(report):
+    """Format the email body for the file integrity report."""
+    lines = []
+
+    for entry in report:
+        lines.append('Message: {}'.format(str(entry.get('message'))))
+        lines.append('Record: {}'.format(entry.get('recid')))
+        lines.append('Report number: {}'.format(entry.get('report_number')))
+        lines.append('Other: {}'.format(entry.get('other')))
+        lines.append('Error message: {}'.format(entry.get('error')))
+        lines.append(('-' * 80) + '\n')
+
+    return '\n'.join(lines)
+
+
+@shared_task
+def subformats_integrity_report():
+    """Send a report of all corrupted subformats to CDS admins."""
+    report = []
+    cache = current_cache.get('subformats_integrity:task_details')
+    task_details = cache if cache else {}
+
+    if 'last_run' not in task_details:
+        task_details['last_run'] = datetime.utcnow()
+
+    record_uuids = (
+        RecordMetadata.query
+        .join(RecordsBuckets)
+        .join(Bucket)
+        .join(ObjectVersion)
+        .filter(ObjectVersion.created > task_details['last_run'])
+        .group_by(RecordMetadata.id).with_entities(RecordMetadata.id)
+        .all())
+
+    records = (CDSRecord.get_record(record.id) for record in record_uuids)
+    for record in records:
+        master = CDSVideosFilesIterator.get_master_video_file(record)
+
+        if not master:
+            report.append({
+                'message': 'No master file',
+                'recid': record['recid'],
+                'report_number': record['report_number'][0]})
+            continue
+
+        master_obj = as_object_version(master['version_id'])
+        try:
+            probe = ff_probe_all(master_obj.file.uri.replace(
+                'root://eosmedia.cern.ch/', ''))
```

Comment:
use [config](https://github.com/CERNDocumentServer/cds-videos/blob/cdslabs/cds/config.py#L835) here, do not hardcode! :)

Reply:
Please add a comment explaining why we do this.
It if because it expects the storage mounted on the machine.